spinalcordtoolbox/straightening.py

```diff
@@ -469,6 +484,8 @@ def straighten(self):
                 distances_curved = centerline.get_distances_from_planes(physical_coordinates,
                                                                         nearest_indexes_curved)
                 lookup = lookup_curved2straight[nearest_indexes_curved]
+                # NB: Any indexes that were mapped to `0` will have their warping field value set to `-100000` (i.e. no warping),
+                #     regardless of the value of `self.threshold_distance`.
                 indexes_out_distance_curved = np.logical_or(
                     np.logical_or(distances_curved > self.threshold_distance,
                                   distances_curved < -self.threshold_distance), lookup == 0)
```

Comment:
Note: I'm kind of suspicious of using the value `0` to mark slice indexes that should be cleared in the warping field.

If there is a value `0` in the lookup table, it's not immediately apparent to me if this means "this index maps to slice index 0, the first z slice of the target image" or if it means "the value for this index has been set to 0 for the purposes of clearing the warping field at this index". _(In fact, does this logic mean that the 0th slice will always get cleared? Are we always ignoring `straight[:, :, 0]` and `curved[:, :, 0]` when constructing the warping field?)_

I'm wondering if it would make more sense to use some other value (`-1`, `None`, etc.) to denote "clear this slice in the warping field", that way we can preserve the actual 0th slice.

Reply:
I agree! In general I would suggest `None`, but it doesn't fit nicely into numpy arrays. I would avoid `-1`, because that's very likely to lead to even weirder indexing bugs (where it may mean the last index). But, it looks like there's absolutely no reason for `lookup_curved2straight` and `lookup_straight2curved` to be converted to numpy arrays, so maybe `None` would work.

I'll let you decide whether this is in-scope for this PR.

Reply:
:100:

Reply:
I'm realizing that this might be a bit time consuming to confirm, and that there are more urgent things I need to work on before the upcoming release (e.g. the PAM50 template issues).

I think I'd rather err on the side of not changing the behavior here, and stick with what has already worked thus far.

Reply:
Issue opened: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4965